tests/unit/test_types.py

```diff
@@ -208,9 +212,405 @@ def test_async_api_spec_from_and_to_dict(faker: Faker):
+        OAuth2Flows(authorization_code=flow)
+
+
+def test_security_scheme_validation():
+    with pytest.raises(ValueError):
+        # missing flows
+        SecurityScheme(type=SecuritySchemesType.OAUTH2)
+
+    with pytest.raises(ValueError):
+        # missing flows
+        SecurityScheme(type=SecuritySchemesType.OPENID_CONNECT)
+
+    with pytest.raises(ValueError):
+        # missing scheme
+        SecurityScheme(type=SecuritySchemesType.HTTP)
+
+    with pytest.raises(ValueError):
+        # missing in
+        SecurityScheme(type=SecuritySchemesType.HTTP_API_KEY)
+    with pytest.raises(ValueError):
+        # invalid in
+        SecurityScheme(type=SecuritySchemesType.HTTP_API_KEY, in_="garbage")
+    with pytest.raises(ValueError):
+        # missing name
+        SecurityScheme(type=SecuritySchemesType.HTTP_API_KEY, in_="header")
+
+
+def test_asyncapi_spec_validation_invalid_security_requirement(faker: Faker):
+    data = {
+        "asyncapi": "2.2.0",
+        "info": {
+            "title": faker.sentence(),
+            "version": faker.pystr(),
+            "description": faker.sentence(),
+        },
+        "channels": {
+            GLOBAL_NAMESPACE: {
+                "description": faker.pystr(),
+                "subscribe": {
+                    "message": {
+                        "oneOf": [
+                            {
+                                "name": faker.pystr(),
+                                "summary": faker.sentence(),
+                                "payload": faker.pydict(value_types=[str, int]),
+                            }
+                            for _ in range(faker.pyint(min_value=2, max_value=10))
+                        ]
+                    }
+                },
+                "publish": {
+                    "message": {
+                        "oneOf": [
+                            {
+                                "title": faker.word(),
+                                "name": faker.pystr(),
+                                "payload": faker.pydict(value_types=[str, int]),
+                                "x-handler": faker.pydict(value_types=[str, int]),
+                                "x-ack": {
+                                    "args": faker.pydict(value_types=[str, int]),
+                                },
+                            }
+                            for _ in range(faker.pyint(min_value=2, max_value=10))
+                        ]
+                    }
+                },
+                "bindings": {
+                    "ws": {
+                        "method": faker.pystr(),
+                        "query": faker.pydict(value_types=[str, int]),
+                    }
+                },
+                "x-handlers": {
+                    "connect": faker.pystr(),
+                    "disconnect": faker.pystr(),
+                    faker.word(): faker.pystr(),
+                },
+            }
+        },
```

Comment:
```suggestion
        "channels": {}
```
Test looks good, but for the sake of making this module shorter you can omit unnecessary data. Same applies to the following tests as well.

Reply:
`info` is also not needed.

Reply:
info was needed, raises an error without it